### ISSUE TITLE
C++11 definition issue

### ISSUE THREAD

**Issue by neheb** [opencv/opencv#15163]
- OpenCV => 4.1.1
- Operating System / Platform = OpenWrt
- Compiler => GCC9 + uClibc-ng

uClibc-ng does not export several C++11 functions into the std namespace. As a result, I currently use this patch: 

```
diff --git a/modules/gapi/include/opencv2/gapi/core.hpp b/modules/gapi/include/opencv2/gapi/core.hpp
index a38d747..f01c0bd 100644
--- a/modules/gapi/include/opencv2/gapi/core.hpp
+++ b/modules/gapi/include/opencv2/gapi/core.hpp
@@ -392,8 +392,8 @@ namespace core {
             {
                 GAPI_Assert(fx != 0. && fy != 0.);
                 return in.withSize
-                    (Size(static_cast<int>(std::round(in.size.width  * fx)),
-                          static_cast<int>(std::round(in.size.height * fy))));
+                    (Size(static_cast<int>(round(in.size.width  * fx)),
+                          static_cast<int>(round(in.size.height * fy))));
             }
         }
     };
diff --git a/modules/gapi/include/opencv2/gapi/own/saturate.hpp b/modules/gapi/include/opencv2/gapi/own/saturate.hpp
index 207dcde..d716643 100644
--- a/modules/gapi/include/opencv2/gapi/own/saturate.hpp
+++ b/modules/gapi/include/opencv2/gapi/own/saturate.hpp
@@ -81,8 +81,8 @@ static inline DST saturate(SRC x, R round)
 // explicit suffix 'd' for double type
 inline double  ceild(double x) { return std::ceil(x); }
 inline double floord(double x) { return std::floor(x); }
-inline double roundd(double x) { return std::round(x); }
-inline double  rintd(double x) { return std::rint(x); }
+inline double roundd(double x) { return round(x); }
+inline double  rintd(double x) { return rint(x); }
 
 } //namespace own
 } //namespace gapi
diff --git a/modules/gapi/src/backends/fluid/gfluidcore.cpp b/modules/gapi/src/backends/fluid/gfluidcore.cpp
index 49d6d2e..e1821b8 100644
--- a/modules/gapi/src/backends/fluid/gfluidcore.cpp
+++ b/modules/gapi/src/backends/fluid/gfluidcore.cpp
@@ -389,7 +389,7 @@ static void run_arithm_s1(uchar out[], const float in[], int width, const float
     cv::util::suppress_unused_warning(v_op);
     for (; w < width; w++)
     {
-        out[w] = saturate<uchar>(s_op(in[w], scalar[0]), std::roundf);
+        out[w] = saturate<uchar>(s_op(in[w], scalar[0]), roundf);
     }
 }
 
@@ -1954,7 +1954,7 @@ GAPI_FLUID_KERNEL(GFluidCartToPolar, cv::gapi::core::GCartToPolar, false)
         {
             float x = in1[l];
             float y = in2[l];
-            float magnitude = std::hypot(y, x);
+            float magnitude = hypot(y, x);
             float angle_rad = std::atan2(y, x);
             float angle = angleInDegrees?
                           angle_rad * static_cast<float>(180 / CV_PI):
```

**Comment by andrey-golubev** [comment]
@dmatveev what do you think?

@neheb please consider opening a PR so we test/look at "live" changes

**Comment by mshabunin** [comment]
@neheb
As I understand uClibc is a C library, why should it export functions in `std::` namespace? What C++ library do you use? Do you build standalone G-API or whole OpenCV?

**Comment by neheb** [comment]
No idea how this works. All I know is, uClibc-ng defines ```__BEGIN_NAMESPACE_STD``` and ```__END_NAMESPACE_STD```

This was the solution in a different project for a slightly different issue: https://github.com/facebook/folly/commit/b2cecfb5ced32cac372cfe6c5ac1935365b5dd60

My C++ is not great for a solution like that.

**Comment by dmatveev** [comment]
ask @elatkin to have a look on this. Wondering if it pass all other builders though

**Comment by elatkin** [comment]
Here we do not necessarily need C++ interface to these math functions.
Accessing through plain C interface must be OK (although need to test):
```c++
// explicit suffix 'd' for double type
inline double  ceild(double x) { return ceil(x); }
inline double floord(double x) { return floor(x); }
inline double roundd(double x) { return round(x); }
inline double  rintd(double x) { return rint(x); }
```
Similarly, it must be OK to call single-precision `hypotf` and `atan2f` here
(single-precision must be faster than double-precision `hypot` and `atan2`):
```c++
float x = in1[l];
float y = in2[l];
float magnitude = hypotf(y, x);
float angle_rad = atan2f(y, x);
```
If using only plain C interface, must be OK to include `math.h` (not `cmath`):
```c++
#include <math.h>
```

**Comment by elatkin** [comment]
@neheb  
Looks like we have two duplicating pull requests, #15211 and #15217  

Could you please review my #15217 ?  

I think it resolves #15163 in the more regular manner:  
- replaces all math function calls with plain C interface  
- uses appropriate precision, e.g. `hypotf` versus `hypot`  
- includes `math.h` instead of `cmath`  

Does such fix make sense for you?

**Comment by neheb** [comment]
Looks fine.
